plugin-enclave-gradle/src/main/kotlin/com/r3/conclave/plugin/enclave/gradle/gramine/GenerateGramineSGXManifest.kt

```diff
@@ -0,0 +1,96 @@
+package com.r3.conclave.plugin.enclave.gradle.gramine
+
+import com.r3.conclave.common.internal.PluginUtils.GRAMINE_ENCLAVE_JAR
+import com.r3.conclave.common.internal.PluginUtils.PYTHON_FILE
+import com.r3.conclave.plugin.enclave.gradle.ConclaveTask
+import org.gradle.api.GradleException
+import org.gradle.api.file.RegularFileProperty
+import org.gradle.api.model.ObjectFactory
+import org.gradle.api.tasks.InputFile
+import org.gradle.api.tasks.Optional
+import org.gradle.api.tasks.OutputFile
+import org.gradle.process.ExecResult
+import java.io.File
+import javax.inject.Inject
+
+open class GenerateGramineSGXManifest @Inject constructor(objects: ObjectFactory) : ConclaveTask() {
+    companion object {
+        const val GRAMINE_SGX_SIGN_EXECUTABLE = "gramine-sgx-sign"
+        const val GRAMINE_GET_TOKEN_EXECUTABLE = "gramine-sgx-get-token"
+    }
+
+    @get:InputFile
+    val privateKey: RegularFileProperty = objects.fileProperty()
+
+    @get:InputFile
+    val directManifest: RegularFileProperty = objects.fileProperty()
+
+    @get:InputFile
+    val enclaveJar: RegularFileProperty = objects.fileProperty()
+
+    @get:InputFile
+    @Optional
+    val pythonSourcePath: RegularFileProperty = objects.fileProperty()
+
+    @get:OutputFile
+    val sgxManifest: RegularFileProperty = objects.fileProperty()
+
+    @get:OutputFile
+    val sgxToken: RegularFileProperty = objects.fileProperty()
+
+    @get:OutputFile
+    val sgxSig: RegularFileProperty = objects.fileProperty()
+
+    override fun action() {
+        val manifestPath = directManifest.get().asFile.absolutePath
+        val privateKeyPath = privateKey.get().asFile.absolutePath
+        val outputSGXManifestPath = sgxManifest.asFile.get().parentFile.absolutePath
+        val enclaveDestinationJarName = "${outputSGXManifestPath}/$GRAMINE_ENCLAVE_JAR"
+        val enclaveDestinationJarFile = File(enclaveDestinationJarName)
+        checkNotNull(
+            enclaveJar.get().asFile.copyTo(
+                enclaveDestinationJarFile,
+                overwrite = true
+            )
+        ) { "Enclave jar file not copied correctly while builgetEnclaveThreadCountFromManifestding the Gramine SGX manifest" }
+        val enclaveDestinationPythonFileName = "${outputSGXManifestPath}/$PYTHON_FILE"
+        val enclaveDestinationPythonFile = File(enclaveDestinationPythonFileName)
+
+        if (pythonSourcePath.isPresent) {
+            checkNotNull(
+                pythonSourcePath.get().asFile.copyTo(
+                    enclaveDestinationPythonFile,
+                    overwrite = true
+                )
+            ) { "Python script not copied correctly while building the Gramine SGX manifest" }
+        }
+
+        if (signDirectManifest(manifestPath, privateKeyPath).exitValue != 0) {
```

Comment:
Gradle by default checks the exit code for you, so there's no need to do this.

Reply:
How is this handled? I see this when I go deeper in the call stack
```
            if (result.exitValue == 0) {
                project.logger.info(execErrorStream.toString().trimEnd())

                // The non-messages have been logged and should be dumped because they are not error messages
                execErrorStream.reset()
            } else if (commandLineConfig.errorOutputStream == null) {

                // Log the error because the callee did not set an error stream
                project.logger.error(execErrorStream.toString().trimEnd())
            }

            return result
```

Reply:
That code uses `project.exec` and you can pass in an `ExecSpec` to configure it. By default it will throw an exception on non-zero exit codes: https://docs.gradle.org/current/javadoc/org/gradle/process/BaseExecSpec.html#isIgnoreExitValue--

Reply:
yes, the above `else` code from `ConclaveTask` will not execute if there's a nonz-zero value and the default for ignoreExitValue is kept to false.

Reply:
Ok, I have just removed my checks then